src/main/scala/is/hail/rvd/OrderedRVD.scala

```diff
@@ -611,7 +563,39 @@ object OrderedRVD {
         Array.tabulate(nPartitions - 1)(i => keys((i + 1) * k))
       }
 
-    UnsafeIndexedSeq(TArray(typ.pkType), rangeBounds)
+    val min = pkis.map(_.min).min(pkOrd)
+    val max = pkis.map(_.max).max(pkOrd)
+
+    OrderedRVDPartitioner.makeRangeBoundIntervals(typ.pkType, min +: partitionMaxes :+ max)
+  }
+
+  def shuffleAndAdjustBounds(typ: OrderedRVDType,
+    partitioner: OrderedRVDPartitioner,
+    rdd: RDD[RegionValue]): OrderedRVD = {
+    if (rdd.isEmpty())
+      return shuffle(typ, partitioner, rdd)
+
+    val pkType = partitioner.pkType
+    val pkOrdUnsafe = pkType.unsafeOrdering(true)
+    val pkOrd = pkType.ordering.toOrdering
+    val pki = getPartitionKeyInfo(typ, OrderedRVD.getKeys(typ, rdd))
+    val min = new UnsafeRow(pkType, pki.map(_.min).min(pkOrdUnsafe))
+    val max = new UnsafeRow(pkType, pki.map(_.max).max(pkOrdUnsafe))
+
+    val newRangeBounds = partitioner.rangeBounds.zipWithIndex.map { case (int: Interval, i: Int) =>
```

Comment:
This is overkill.  I think you can just do:

```
  val newRangeBounds = partitioner.rangeBounds.toArray
  newRangeBounds(0) = newRangeBounds(0).copy(start = min...)
  newRangeBounds(newRangeBounds.length - 1) = ... max ...  
```

Also calling a non-integer `int` isn't preferred.